components/generic/CompareImageSlider.vue

```diff
@@ -0,0 +1,224 @@
+      @mousedown="initDrag"
+    >
+      <span class="slider-bar" />
+      <button
+        :class="{ 'is-active' : dragging }"
+        class="slider-handle"
+        data-qa="compare image slider handler"
+      >
+        <span class="sr-only">Slider Handle</span>
+      </button>
+    </div>
+  </figure>
+</template>
+
+<script>
+  export default {
+    name: 'CompareImageSlider',
+
+    props: {
+      imageLeft: {
+        type: String,
+        required: true
+      },
+
+      imageRight: {
+        type: String,
+        required: true
+      },
+
+      imageLeftText: {
+        type: String,
+        default: ''
+      },
+
+      imageRightText: {
+        type: String,
+        default: ''
+      }
+    },
+
+    data() {
+      return {
+        dragging: false,
+        imageWidth: 0,
+        sliderWidth: 0,
+        sliderPosition: 0.5
+      };
+    },
+
+    computed: {
+      leftImageClip() {
+        const rightLength = this.imageWidth * this.sliderPosition;
+
+        return {
+          clip: `rect(auto, ${rightLength}px, auto, auto)`
+        };
+      },
+
+      sliderBarPosition() {
+        const leftPosition = this.imageWidth * this.sliderPosition - this.sliderWidth / 2;
+        return {
+          left: `${leftPosition}px`
+        };
+      }
+    },
+
+    mounted() {
+      this.setImageWidth();
+      this.setSliderWidth();
+
+      window.addEventListener('resize', this.setImageWidth);
+      window.addEventListener('mousemove', this.drag);
+      window.addEventListener('mouseup', this.stopDrag);
+    },
+
+    methods: {
+      setImageWidth() {
+        this.imageWidth = this.$refs.imageRight.getBoundingClientRect().width;
+      },
+
+      setSliderWidth() {
+        this.sliderWidth = this.$refs.slider.getBoundingClientRect().width;
+      },
+
+      initDrag() {
+        this.dragging = true;
+      },
+
+      stopDrag() {
+        this.dragging = false;
+      },
+
+      drag(event) {
+        // Calc Cursor Position from the left edge of the viewport
+        const cursorXfromViewport = event.pageX;
+        // Calc Cursor Position from the left edge of the window (consider any page scrolling)
+        const cursorXfromWindow = cursorXfromViewport - window.pageXOffset;
+        // Calc Cursor Position from the left edge of the image
+        const imagePosition = this.$refs.imageRight.getBoundingClientRect();
+        let pos = cursorXfromWindow - imagePosition.left;
+
+        const minPos = 0 + this.sliderWidth / 2;
+        const maxPos = this.imageWidth - this.sliderWidth / 2;
+
+        if (pos < minPos) {
+          pos = minPos;
+        }
+
+        if (pos > maxPos) {
+          pos = maxPos;
+        }
+
+        if (this.dragging) {
+          this.sliderPosition = pos / this.imageWidth;
+        }
+      }
+    }
+  };
+</script>
+
+<style lang="scss" scoped>
+  $slider-width: 48px;
+
+  .compare-image {
+    position: relative;
+    overflow: hidden;
```

Comment:
It might be good to introduce a height or max-height, for example if images are of different sizes and to avoid having a compare thing that is too large in height.

Reply:
Should we compare images that do not have the same dimensions? @rwd What are your thoughts?

Reply:
We'll need to handle different size images. They may not always be drastically different dimensions, but should always be made to fit.